project/src/com/google/daggerqueryui/page.html

```diff
@@ -25,15 +25,60 @@
     <script src="third-party/js/jquery.min.js"></script>
     <!-- MDB core JavaScript -->
     <script type="text/javascript" src="third-party/js/mdb.min.js"></script>
+    <!-- Bootstrap tooltips -->
+    <script type="text/javascript" src="third-party/js/popper.min.js"></script>
+    <!-- Bootstrap core JavaScript -->
+    <script type="text/javascript" src="third-party/js/bootstrap.min.js"></script>
 </head>
 <body>
 
 <div class="container" id="query-text-field">
-    <div class="md-form form-lg">
-        <input type="text" id="inputLGEx" class="form-control form-control-lg" spellcheck="false">
-        <label for="inputLGEx">Your query</label>
+    <div class="row align-items-center">
+        <div class="col-11">
+            <div class="md-form form-lg">
+                <input type="text"
+                       id="inputLGEx"
+                       class="form-control form-control-lg"
+                       spellcheck="false">
+                <label for="inputLGEx">
+                    Your query
+                </label>
+            </div>
+        </div>
+        <div class="col-1" style="background: #FFF;">
+            <a tabindex="0"
+               id="help-button"
+               popover-content="#instructions"
+               data-toggle="popover"
+               data-trigger="active">
+                <img src="res/help-button-icon.svg"/>
+            </a>
+
+            <div class="hidden" id="instructions">
+                <div class="popover-heading">
+                    How to use <b>Dagger Query UI</b>?
+                </div>
+                <div class="popover-body">
+                    You can execute the following queries by entering them in the input field:
+                    <ul>
+                        <li>
+                            <b class="query-name">deps</b>
+                            <b>binding</b> - to see all nodes dependencies
+                        </li>
+                        <li>
+                            <b class="query-name">allpaths</b>
+                            <b>source target</b> - to see all paths between source and target nodes
+                        </li>
+                        <li>
+                            <b class="query-name">somepath</b>
+                            <b>source target</b> - to see some path between source and target nodes
+                        </li>
+                    </ul>
+                </div>
+            </div>
+        </div>
     </div>
 </div>
-
 </body>
+<script type="text/javascript" src="scripts/enables-popover.js"></script>
```

Comment:
Did you mean to put this inside `<body></body>`? See https://stackoverflow.com/a/31867782

Reply:
Interesting, thanks. Yes, let's put it in the <body> tag, because implicitly creating another tag sounds very bad.